.github/workflows/test-ci.yml

```diff
@@ -282,7 +280,7 @@ jobs:
   BuildDocs:
     name: Build API Documentation
     runs-on: gpu
-    if: ${{ (contains(github.event.head_commit.message, 'docs_build')) || (github.ref == 'refs/heads/master')}}
+    if: ${{ !(contains(github.event.head_commit.message, 'docs_build')) || (github.ref == 'refs/heads/master') }}
```

Comment:
so whats the expected behavior ? one needs to put `docs_build` to skip the building of the documentation ? seems weird to me...

Reply:
Its weird, when I added this, it is building .. I am lost but it works :P

Reply:
Let's clear this then, We should build the doc: 
- In PR to master, when the `docs_build` keyword is present.
- On every commit to master (aka after merge)

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts#github-context 

I think we should use ``github.head_ref`` (and the contains)

Reply:
The ref aspect is fine. The containts part doesnt work. I am trying to understand whats broken. For some reason it does not trigger even when I use `docs_build`

Reply:
Okay this is broken. For now, we dont check `docs_build` and also `style`. (Cause `github.event.head_commit` is empty if the event is triggered with `pull_request` I will open an issue to fix this, but its not easy :P 
We can proceed with current PR for now,.